modal/cls.py

```diff
@@ -90,7 +90,7 @@ def _uses_common_service_function(self):
 
     def __init__(
         self,
-        user_cls: type,
+        user_cls: Optional[type],  # this would be None in case of lookups
```

Comment:
Just a random comment about consistency; I think the analogous attribute is `raw_f` for `Function`?

Reply:
Yup. There is a bunch of code here that needs cleanup/clarification. `FunctionInfo` is quite annoying how it bundles a bunch of different use cases but is hard to provide in some cases.